pkg/controller/gitopsconfig/controller.go

```diff
@@ -414,24 +414,23 @@ func (r *Reconciler) manageDeletion(instance *gitopsv1alpha1.GitOpsConfig) (reco
 		//we return because we need to wait for the job to stop
 		return reconcile.Result{
 			Requeue:      true,
-			RequeueAfter: time.Minute,
+			RequeueAfter: time.Second * 12,
 		}, nil
 	}
 	//There should be only one pending job
 	job := applicableJobList[0]
-	if job.Status.Succeeded > 0 {
+	if job.Status.Succeeded > 0 || job.Status.Failed >= *job.Spec.BackoffLimit {
```

Comment:
As we discussed, I'd be grateful if you could remove the finalizer cleanup in case Job has failed. I believe it's better (if uglier) to still hang (so that user knows there's something wrong), than to hide the failure under the rug (and leave the user oblivious to the fact stuff didn't get deleted).